### PR TITLE
Adds changes from swan-integration branch

### PR DESCRIPTION
**Pull request by techtocore**
Signed-off-by: Akash Ravi <akashkravi@gmail.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,7 +10,8 @@ rm -rf /opt/conda/lib/python3.7/site-packages/notebook/templates && \
 cp -rl templates /opt/conda/lib/python3.7/site-packages/notebook/ && \
 ls && \
 cd SwanContents && pip install --no-deps . && \
-jupyter nbextension install --py --system swancontents
+jupyter nbextension install --py swancontents --system --symlink && \
+jupyter nbextension enable swancontents --system --py
```
Comment (diocas) [inline Dockerfile]: `swancontents` should not be enabled

File: extension/js/scripts.js [hunk 2]
```diff
@@ -26,6 +31,9 @@ function package_view(dir) {
 
         var toInstall = [];
         views.select_to_install(toInstall);
+
+        $('#packageview').show();
+        $('#loadingview').hide();
     });
```
Comment (diocas) [inline extension/js/scripts.js:39]: I see 2 problems:
1. You show if it succeeds, right? And if it fails? Will it continue to show the loading without telling anything to the user and without showing the list/add pkgs view again?
1. And if it fails (i.e network connectivity), will the user have to add new packages to install again, from scratch? (for this problem I don't have a strong opinion. Let's keep it as is)

**Review by diocas — CHANGES_REQUESTED** [review]
Some comments

File: extension/js/scripts.js [hunk 3]
```diff
@@ -34,6 +33,9 @@ function package_view(dir) {
 
         $('#packageview').show();
         $('#loadingview').hide();
+    }, function () {
+        common.display_msg("Server Error", "The project configuration could not be retrived. Please try again.");
```
Comment (diocas) [inline extension/js/scripts.js:37]: How can I try again? Closing and clicking again in the cog?

Comment (techtocore) [reply extension/js/scripts.js:37]: Yes. Is that okay?

Comment (diocas) [reply extension/js/scripts.js:37]: Ok, let's leave it as is. I'm gonna merge and test everything again and create an issue with my observations, ok?

**Pull request merged**